### PR TITLE
Sequence elementsEqual matcher

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
The PR should summarize what was changed and why. Here are some questions to
help you if you're not sure:

 - What behavior was changed? None, only additional changes.
 - What code was refactored / updated to support this change? None
 - What issues are related to this PR? Or why was this change introduced? None

Checklist - While not every PR needs it, new features should consider this list:

 - [x] Does this have tests?
 - [x] Does this have documentation?
 - [x] Does this break the public API (Requires major version bump)?
 - [x] Is this a new feature (Requires minor version bump)?

Hey guys :))
This PR is a proposal for a Sequence::elementsEqual matcher.
Let me know your thoughts on that 😊

### CODE REVIEW

**Comment by LucianoPAlmeida** [comment]
Hey @ikesyo :))
Sure, extracted 👍

File: Tests/NimbleTests/XCTestManifests.swift [hunk 1]
```diff
@@ -231,6 +231,12 @@ extension EqualTest {
     ]
 }
 
+extension ElementsEqualTest {
+    static var __allTests = [
```
Comment (wongzigii) [inline Tests/NimbleTests/XCTestManifests.swift]: Prefer `let` over `var`

File: Tests/NimbleTests/Matchers/ElementsEqualTests.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+//  ElementsEqualTests.swift
+//  Nimble
+//
+//  Created by Luciano Almeida on 7/1/18.
+//  Copyright © 2018 Jeff Hui. All rights reserved.
+//
+
+import Foundation
+
+import Foundation
```
Comment (wongzigii) [inline Tests/NimbleTests/Matchers/ElementsEqualTests.swift]: `import Foundation` is duplicate

File: Tests/NimbleTests/Matchers/ElementsEqualTests.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import XCTest
+import Nimble
+
+final class ElementsEqualTest: XCTestCase, XCTestCaseProvider {
+
+    func testSequenceElementsEqual() {
```
Comment (wongzigii) [inline Tests/NimbleTests/Matchers/ElementsEqualTests.swift]: Prefer function name -> `testSequenceElementsEquality`, just like functions name in `EqualTest`.

**Comment by LucianoPAlmeida** [comment]
Hey @wongzigii :))
Fixed, thanks for the comments 👍

File: Nimble.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -613,6 +619,8 @@
 		AE4BA9AC1C88DDB500B73906 /* Errors.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = Errors.swift; sourceTree = "<group>"; };
 		AE7ADE441C80BF8000B94CD3 /* MatchError.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = MatchError.swift; sourceTree = "<group>"; };
 		AE7ADE481C80C00D00B94CD3 /* MatchErrorTest.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = MatchErrorTest.swift; sourceTree = "<group>"; };
+		B20058C020E92C7500C1264D /* ElementsEqual.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ElementsEqual.swift; sourceTree = "<group>"; };
+		B20058C420E92CE400C1264D /* ElementsEqualTests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ElementsEqualTests.swift; sourceTree = "<group>"; };
```
Comment (ikesyo) [inline Nimble.xcodeproj/project.pbxproj]: This should be named `ElementsEqualTest.swift` for consistency.

File: Sources/Nimble/Matchers/ElementsEqual.swift [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+//  Nimble
+//
+//  Created by Luciano Almeida on 7/1/18.
+//  Copyright © 2018 Jeff Hui. All rights reserved.
+//
+
+/// A Nimble matcher that succeeds when the actual sequence contain the same elements in the same order to the exepected sequence.
+public func elementsEqual<S: Sequence>(_ expectedValue: S?) -> Predicate<S> where S.Element: Equatable {
+    // A matcher abstraction for https://developer.apple.com/documentation/swift/sequence/2949668-elementsequal
+    return Predicate.define("elementsEqual <\(stringify(expectedValue))>", matcher: { (actualExpression, msg) in
```
Comment (ikesyo) [inline Sources/Nimble/Matchers/ElementsEqual.swift]: > `, matcher: { (actualExpression, msg) in`

Could you please use trailing closure syntax here instead?

File: Sources/Nimble/Matchers/ElementsEqual.swift [hunk 6]
```diff
@@ -0,0 +1,24 @@
+//
+//  ElementsEqual.swift
+//  Nimble
+//
+//  Created by Luciano Almeida on 7/1/18.
+//  Copyright © 2018 Jeff Hui. All rights reserved.
```
Comment (ikesyo) [inline Sources/Nimble/Matchers/ElementsEqual.swift]: Could you remove these header comments as same as other source files?

**Comment by LucianoPAlmeida** [comment]
Hey @ikesyo 
Sure, fixed 👍 
Thanks for the comments :))

**Pull request merged**